definitions/features/foreman_proxy.rb

```diff
@@ -3,19 +3,62 @@ class Features::ForemanProxy < ForemanMaintain::Feature
     label :foreman_proxy
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  attr_reader :dhcpd_conf_file, :cert_path, :request_url,
+              :default_ssl_cert_options
 
   def initialize
     @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
     @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+    @request_url = "https://#{hostname}:9090/dhcp"
+    @default_ssl_cert_options = load_default_ssl_cert_options
   end
 
   def valid_dhcp_configs?
     dhcp_req_pass? && !syntax_error_exists?
   end
 
+  def list_reservations(subnet)
+    resp = dhcp_restclient["/#{subnet}"].get { |response, _request, _result| response }
```

Comment:
@iNecas ,

I need your suggestion for these API requests.
Whether it would be good - instead of placing API calls in `foreman_proxy` feature If I extract out these API calls into `utils/api` folder to add class files in which it will do API interface.

for example, in this case I will add dhcp_reservation.rb under utils/api folder to add methods fetch_list/add/delete.


```
module ForemanMaintain
  module Utils
    module Api
      class DhcpReservation
        include ForemanMaintain::Concerns::SystemHelpers

        attr_reader :dhcp_api_resource

        def self.instance(base_url, options = {})
          @instance ||= new(base_url, options)
        end

        def initialize(base_url, options = {})
          @dhcp_api_resource = RestClient::Resource.new(base_url, options)
        end

        def fetch_list(subnet)
          resp = dhcp_api_resource["/#{subnet}"].get { |response, _request, _result| response }
          handle_api_response(resp, true)
        end
     end
    end
end
```